src/FakeItEasy/Core/SequenceNumberManager.cs

```diff
@@ -1,14 +1,29 @@
 ﻿namespace FakeItEasy.Core
 {
+    using System.Runtime.CompilerServices;
     using System.Threading;
 
     internal static class SequenceNumberManager
     {
-        private static int sequenceNumber;
+        private static readonly ConditionalWeakTable<ICompletedFakeObjectCall, StrongBox<int>> SequenceNumbers =

```

Comment:
It's used to box a value type in a strongly typed manner (as the name implies). I discovered it by chance a long time ago, but I had never used it until now.
(I can't use `int` directly, because `ConditionalWeakTable<TKey, TValue>` has a `class` constraint on `TValue`)